File: problem0001.java
public class problem0001 {
    public static void main(String[] args) {
        int[] nums = new int[]{2,7,11,15};
        int target = 9;
        int[] result = twoSum(nums, target);    // result = [0, 1] => 2 + 7 = 9
        System.out.printf("[%d, %d]\n", result[0], result[1]);
    }

    public static int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        for(int i = 0; i < nums.length; i++) {
            int firstNum = nums[i];
            int secondNum = target - firstNum;
            for(int j = i + 1; j < nums.length; j++) {
                if(nums[j] == secondNum) {
                    result[0] = i;
                    result[1] = j;
                    return result;
                }
            }
        }
        return result;
    }
}

File: LeetCodeGo/problem_solution/problem0001.go
package main

func twoSum(nums []int, target int) []int {
	result := make([]int, 2)
	for i := 0; i < len(nums); i++ {
		firstNum := nums[i]
		secondNum := target - firstNum
		for j := i + 1; j < len(nums); j++ {
			if nums[j] == secondNum {
				result[0] = i
				result[1] = j
				return result
			}
		}
	}

	return result
}
